File: sgf-poker/sgf-poker-ios/Core/UseCases/FetchPlayersUseCase.swift
// Core/UseCases/FetchPlayersUseCase.swift

/// Retrieves the full ordered player list from the repository.
final class FetchPlayersUseCase {

    private let repository: PlayerRepository

    init(repository: PlayerRepository) {
        self.repository = repository
    }

    func execute() throws -> [Player] {
        try repository.fetchAll()
    }
}

File: sgf-poker-android/app/src/main/java/com/sgf/poker/usecases/player/FetchPlayersUseCase.java
package com.sgf.poker.usecases.player;

import com.sgf.poker.data.repository.PlayerRepository;
import com.sgf.poker.domain.model.Player;
import java.util.List;

/** Retrieves all players sorted alphabetically. */
public class FetchPlayersUseCase {

    private final PlayerRepository repository;

    public FetchPlayersUseCase(PlayerRepository repository) {
        this.repository = repository;
    }

    public List<Player> execute() {
        return repository.fetchAll();
    }
}
